lib/chef/resource/runit_service.rb

```diff
@@ -0,0 +1,693 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require "chef/resource"
+require "chef/resource/service"
+
+class Chef
+  class Resource
+    # Missing top-level class documentation comment
+    class RunitService < Chef::Resource::Service
+      resource_name :runit_service
+
+      description "The runit_service manages services running under the runit init system. Note: The runit init system must be installed to use this resource."
+
+      introduced "15.2"
+
+      # For legacy reasons we allow setting these via attribute
+      property :sv_bin, String,
+        default: lazy { (node["runit"] && node["runit"]["sv_bin"]) || (platform_family?("debian") ? "/usr/bin/sv" : "/sbin/sv") },
+        description: "The path to the sv program binary. This will attempt to use the node['runit']['sv_bin'] attribute, and falls back to /usr/bin/sv."
+
+      property :sv_dir, [String, FalseClass],
+        default: lazy { (node["runit"] && node["runit"]["sv_dir"]) || "/etc/sv" },
+        description: "The base 'service directory' for the services managed by the resource. By default, this will attempt to use the node['runit']['sv_dir'] attribute, and falls back to /etc/sv."
+
+      property :service_dir, String,
+        default: lazy { (node["runit"] && node["runit"]["service_dir"]) || "/etc/service" },
+        description: "The directory where services are symlinked to be supervised by runsvdir. By default, this will attempt to use the node['runit']['service_dir'] attribute, and falls back to /etc/service."
+
+      property :lsb_init_dir, String,
+        default: lazy { (node["runit"] && node["runit"]["lsb_init_dir"]) || "/etc/init.d" },
+        description: "The directory where an LSB-compliant init script interface will be created. By default, this will attempt to use the node['runit']['lsb_init_dir'] attribute, and falls back to /etc/init.d."
+
+      property :control, Array, default: [],
+               description: "An array of signals to customize control of the service, see runsv man page on how to use this. This requires that each template be created with the name sv-service_name-signal.erb."
+
+      property :env, Hash, default: {},
+               description: "A hash of environment variables with their values as content used in the service's env directory. Default is an empty hash. When this hash is non-empty, the contents of the runit service's env directory will be managed by Chef in order to conform to the declared state."
+
+      property :options, Hash, # deprecated: true should be set in the future
+        default: lazy { default_options }, coerce: proc { |r| default_options.merge(r) if r.respond_to?(:merge) },
+        description: "Options passed as variables to templates, for compatibility with legacy runit service definition. Default is an empty hash."
```

Comment:
The default in the description ("an empty hash") doesn't seem to match the default in line 56. Does something need to be modified?